datamodel_code_generator/parser/openapi.py

```diff
@@ -274,7 +274,11 @@ def get_ref_model(self, ref: str) -> Dict[str, Any]:
 
     def parse_parameters(self, parameters: ParameterObject, path: List[str]) -> None:
         if parameters.name and parameters.schema_:  # pragma: no cover
-            self.parse_item(parameters.name, parameters.schema_, [*path, 'schema'])
+            data_model_type = self.parse_item(
+                parameters.name, parameters.schema_, [*path, 'schema']
+            )
+            if OpenAPIScope.Parameters in self.open_api_scopes:
+                self.results.append(data_model_type)
```

Comment:
@hambrosia  @pvicol 
`self.parse_item` returns `DataType`.But, `self.results` is list of `DataModel`